contrib/parseq-retry/src/main/java/com/linkedin/parseq/retry/AbstractRetryPolicy.java

```diff
@@ -0,0 +1,146 @@
+package com.linkedin.parseq.retry;
+
+import com.linkedin.parseq.retry.backoff.BackoffPolicy;
+import com.linkedin.parseq.retry.monitor.EventMonitor;
+import com.linkedin.parseq.retry.termination.TerminationPolicy;
+
+import java.util.function.Function;
+
+
+/**
+ * A base class for policies that enable customizable retries for arbitrary parseq tasks.
+ *
+ * @param <T> Type of a task result, used for strongly typed processing of outcomes.
+ *
+ * @author Oleg Anashkin (oleg.anashkin@gmail.com)
+ */
+public abstract class AbstractRetryPolicy<T> {
```

Comment:
The best practice we've been using so far is to define both an interface and a builder class - the interface will have all the getters and builder will have all the setters. You can either have a separate impl class or combine it with your builder, but make sure you don't leak the constructor. That way you are free to make incremental changes to your interface in the future without breaking backward compatibility.